### PR TITLE
HGcal Hexgonal Geometry

### PR DESCRIPTION
**Pull request by ianna**
- Merge Sunanda's bsunanda:Run2-hcx60 branch
- Fix conflicts
- Move HGCalParameters class to common HGCal directory
- Add a schema for HGCalParameters as PHGCalParameters
- Add serialization test for a new schema PHGCalParameters

Replaces #12845

### CODE REVIEW

**Comment by ianna** [comment]
@cmsbuild please test

**Comment by ianna** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/12862	95fcf83
- from reco perspective, this is equivalent to already signed https://github.com/cms-sw/cmssw/pull/12845	 d51b973
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by ianna** [comment]
@slava77 - this PR is the same as you've already signed (just moving files around).
@ggovi and @diguida - this is a first iteration of a new schema for HGCal parameters. Please, let me know if you have time to look at it. I'd like to integrate the PR ASAP as it involves changes in DataFormats. Alternatively, I can remove the schema from this PR to speed up its integration. Thanks.

**Comment by mmusich** [comment]
+1 
adding `PHGCalParameters` CondFormat

**Comment by mmusich** [comment]
@ggovi please have a look too

**Comment by ggovi** [comment]
+1

**Comment by ianna** [comment]
@civanch - could you, please, have a look? Thanks.

File: Geometry/HGCalCommonData/data/v1/hgcalEE.xml [hunk 1]
```diff
@@ -110,6 +110,7 @@
   <SpecPar name="HGCalEE">
     <PartSelector path="//HGCalEESensitive.*"/>
     <Parameter name="Volume" value="HGCalEESensitive" eval="false"/>
+    <Parameter name="GeometryMode" value="HGCalGeometryMode::Square" eval="false"/>
```
Comment (davidlange6) [inline Geometry/HGCalCommonData/data/v1/hgcalEE.xml]: @ianna - do I understand this code correctly to say that in each xml we have a line to which "GeometryMode" we do _NOT_ use? Could we not make this more descriptive by giving the GeometryMode we DO use instead?

Comment (ianna) [inline Geometry/HGCalCommonData/data/v1/hgcalEE.xml]: @davidlange6 - your assumption is not correct. HGCal "GeometryMode" is used starting from this PR. It is either a square or a hexagon reflecting its sensors shape.

Comment (davidlange6) [inline Geometry/HGCalCommonData/data/v1/hgcalEE.xml]: ok Thanks!

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**